syntax/format.go

```diff
@@ -0,0 +1,115 @@
+package syntax
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/arr-ai/arrai/rel"
+)
+
+// FormatString returns a string which represents `rel.Value` with more reabable format.
+// For example, `{b: 2, a: 1, c: (a: 2, b: {aa: {bb: (a: 22, d: {3, 1, 2})}})}` can be formatted to:
+//{
+//	b: 2,
+//	a: 1,
+//	c: (
+//		a: 2,
+//		b: {
+//			aa: {
+//				bb: (
+//					a: 22,
+//					d: {
+//						3,
+//						1,
+//						2
+//					}
+//				)
+//			}
+//		}
+//	)
+//}
+func FormatString(val rel.Value, identsNum int) string {
+	identsNum = identsNum + 1
+	switch t := val.(type) {
+	case rel.Tuple: // (a: 1)
+		return formatTupleString(t, identsNum)
+	case rel.Dict: // {'a': 1}
+		return formatDictString(t, identsNum)
+	case rel.Array:
+		return t.String()
+	case rel.Set: // {1, 2}
+		return formatSetString(t, identsNum)
+	default:
+		return t.String()
+	}
+}
+
+func formatTupleString(tuple rel.Tuple, identsNum int) string {
+	var sb strings.Builder
+	identsStr := getIdents(identsNum)
+	sb.WriteString("(")
+	for index, enum := 0, tuple.Enumerator(); enum.MoveNext(); index++ {
+		name, val := enum.Current()
+		fmt.Fprintf(&sb, getFormat(index, tuple.Count()), identsStr, name, FormatString(val, identsNum))
+	}
+
+	sb.WriteString(fmt.Sprintf("%s)", getIdents(identsNum-1)))
+	return sb.String()
+}
+
+func formatDictString(dict rel.Dict, identsNum int) string {
+	var sb strings.Builder
+	identsStr := getIdents(identsNum)
+	sb.WriteString("{")
+	for index, enum := 0, dict.DictEnumerator(); enum.MoveNext(); index++ {
+		key, val := enum.Current()
+		fmt.Fprintf(&sb, getFormat(index, dict.Count()), identsStr, key, FormatString(val, identsNum))
+	}
+
+	sb.WriteString(fmt.Sprintf("%s}", getIdents(identsNum-1)))
+	return sb.String()
+}
+
+func formatSetString(set rel.Set, identsNum int) string {
+	var sb strings.Builder
+	identsStr := getIdents(identsNum)
+	sb.WriteString("{")
+	for index, enum := 0, set.Enumerator(); enum.MoveNext(); index++ {
+		format := ",\n%s%v"
+		if index == 0 && set.Count() == 1 {
+			format = format[1:] + "\n"
+		} else if index == 0 && set.Count() > 1 {
+			format = format[1:]
+		} else if index == set.Count()-1 && set.Count() > 1 {
+			format = format + "\n"
+		}
+
+		val := enum.Current()
+		fmt.Fprintf(&sb, format, identsStr, FormatString(val, identsNum))
+	}
+
+	sb.WriteString(fmt.Sprintf("%s}", getIdents(identsNum-1)))
+	return sb.String()
+}
+
+func getFormat(index, length int) string {
+	format := ",\n%s%v: %v"
+	if index == 0 && length == 1 {
+		format = format[1:] + "\n"
+	} else if index == 0 && length > 1 {
+		format = format[1:]
+	} else if index == length-1 && length > 1 {
+		format = format + "\n"
+	}
+
+	return format
+}
+
+func getIdents(identsNum int) string {
+	var sb strings.Builder
+	for i := 0; i < identsNum; i++ {
+		sb.WriteString("\t")
```

Comment:
Nit: Can we use two spaces rather than a tab? arr.ai data can get deeply nested quickly, and once the indents get longer than a screen and have to wrap the pretty printing will be almost useless, so two spaces should keep the indents smaller.

Reply:
Another option would be to make the indent an input so the user can change it, kinda like [JSON.stringify](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify).

However I wouldn't recommend it at this point. Let's see how this feels to use first.

Reply:
Changed tab to two spaces, and current result looks like:
```
@> //fmt.pretty({'a':1, 'b':2, 'c':(d:11, e:22, f:{111, 222})})
{
  b: 2,
  a: 1,
  c: (
    d: 11,
    f: {
      111,
      222
    },
    e: 22
  )
}
```